### PR TITLE
feat: Add Vertex AI Realtime provider implementing IRealtimeClient/IRealtimeClientSession

### PR DESCRIPTION
**Pull request by tarekgh**
## Summary

Adds a Vertex AI Live API provider implementing the `Microsoft.Extensions.AI` Realtime abstractions (`IRealtimeClient` / `IRealtimeClientSession`), enabling real-time audio, text, image, and function-calling conversations with Vertex AI models through the standardized MEAI interface.

This follows the same pattern as the [Gemini Realtime provider PR](https://github.com/googleapis/dotnet-genai/pull/256) in the `dotnet-genai` repository, and is consistent with the existing `IChatClient` implementation (`PredictionServiceChatClient`) in this package.

### AOT Compatibility

This PR also includes cross-cutting AOT (Ahead-of-Time) compilation improvements that span the entire SDK, not just the realtime provider:

**Realtime provider AOT support:**
- Added `InternalLiveJsonContext` — a source-generated `JsonSerializerContext` with `[JsonSerializable]` entries for all Live API types (including `Dictionary<string, object?>` for function call arguments)
- All WebSocket JSON serialization/deserialization uses the source-generated context via `LiveJsonContext.Default.LiveClientMessage` / `LiveJsonContext.Default.LiveServerMessage`
- All internal types use explicit `[JsonPropertyName]` attributes — no reflection-based naming
- Added `AotJsonContextTest.cs` verifying source-gen coverage, nested type auto-discovery, round-trip correctness, and `DefaultIgnoreCondition.WhenWritingNull` behavior

**Chat client AOT improvement (`PredictionServiceChatClient.cs`):**
- Eliminated JSON round-tripping for tool arguments and results: previously, `Struct` values were converted via `Struct.Parser.ParseJson(JsonSerializer.Serialize(value))` — a serialize-then-parse pattern that relies on reflection-based `JsonSerializer.Serialize`. Now uses direct `Struct` ↔ dictionary conversion, avoiding `System.Text.Json` entirely on the tool-call hot path. This makes the existing `IChatClient` more AOT-friendly as a side effect.

## Usage Example

```csharp
using Google.Cloud.AIPlatform.V1;
using Google.Cloud.VertexAI.Extensions;
using Microsoft.Extensions.AI;

// Create the realtime client from a PredictionServiceClientBuilder
var builder = new PredictionServiceClientBuilder
{
    // Credentials are resolved automatically via ADC, or set explicitly:
    // JsonCredentials = File.ReadAllText("service-account.json"),
};

IRealtimeClient realtimeClient = builder.BuildIRealtimeClient(
    "gemini-live-2.5-flash-native-audio");

// Or async:
// IRealtimeClient realtimeClient = await builder.BuildIRealtimeClientAsync(
//     "gemini-live-2.5-flash-native-audio");

// Define a tool for function calling
AIFunction getWeather = AIFunctionFactory.Create(
    (string location) =>
        location.ToLowerInvariant() switch
        {
            var l when l.Contains("seattle")       => $"The weather in {location} is rainy, 55F",
            var l when l.Contains("new york")      => $"The weather in {location} is cloudy, 70F",
            var l when l.Contains("san francisco") => $"The weather in {location} is foggy, 60F",
            _                                      => $"Sorry, I don't have weather data for {location}."
        },
    "GetWeather",
    "Gets the current weather for a given location");

// Configure session options
var sessionOptions = new RealtimeSessionOptions
{
    Instructions = "You are a helpful assistant.",
    Voice = "Puck",
    OutputModalities = ["audio"],
    Tools = [getWeather],
    InputAudioFormat = new RealtimeAudioFormat("audio/pcm;rate=16000", 16000),
    TranscriptionOptions = new TranscriptionOptions(),
    VoiceActivityDetection = new VoiceActivityDetectionOptions
    {
        Enabled = true,
        AllowInterruption = true,
    },
};

// Optionally wrap with function invocation middleware
var clientBuilder = new RealtimeClientBuilder(realtimeClient)
    .UseFunctionInvocation();
IRealtimeClient wrappedClient = clientBuilder.Build();

// Create a session and start streaming
await using var session = await wrappedClient.CreateSessionAsync(sessionOptions);

// Start listening for server messages in the background
_ = Task.Run(async () =>
{
    await foreach (var message in session.GetStreamingResponseAsync(cancellationToken))
    {
        switch (message)
        {
            case OutputTextAudioRealtimeServerMessage audio
                when audio.Type == RealtimeServerMessageType.OutputAudioDelta:
                // Play audio - audio.Audio contains base64-encoded PCM audio
                PlayAudio(Convert.FromBase64String(audio.Audio));
                break;

            case OutputTextAudioRealtimeServerMessage text
                when text.Type == RealtimeServerMessageType.OutputAudioTranscriptionDelta:
                Console.Write(text.Text);
                break;

            case InputAudioTranscriptionRealtimeServerMessage transcription:
                Console.WriteLine($"You said: {transcription.Transcription}");
                break;

            case ResponseCreatedRealtimeServerMessage response:
                if (response.Usage != null)
                    Console.WriteLine(
                        $"Tokens - In: {response.Usage.InputTokenCount}, " +
                        $"Out: {response.Usage.OutputTokenCount}");
                break;

            case ErrorRealtimeServerMessage error:
                Console.Error.WriteLine($"Error: {error.Error?.Message}");
                break;
        }
    }
});

// Send audio from microphone (e.g., 16kHz PCM)
var audioContent = new DataContent(
    $"data:audio/pcm;base64,{Convert.ToBase64String(audioBytes)}");
await session.SendAsync(
    new InputAudioBufferAppendRealtimeClientMessage(audioContent));
await session.SendAsync(
    new InputAudioBufferCommitRealtimeClientMessage());
await session.SendAsync(
    new CreateResponseRealtimeClientMessage());

// Send an image for multimodal conversation
var imageContent = new DataContent(
    $"data:image/jpeg;base64,{Convert.ToBase64String(imageBytes)}");
var imageItem = new RealtimeConversationItem(
    [imageContent], id: null, role: ChatRole.User);
await session.SendAsync(
    new CreateConversationItemRealtimeClientMessage(item: imageItem));
await session.SendAsync(
    new CreateResponseRealtimeClientMessage());
```

### New Files
- **`PredictionServiceRealtimeClient.cs`** — `IRealtimeClient` implementation that wraps a `PredictionServiceClientBuilder`, resolves credentials (ADC, service account JSON, scoped OAuth), builds the WebSocket connection, and creates realtime sessions via the Vertex AI Live API.
- **`PredictionServiceRealtimeSession.cs`** — `IRealtimeClientSession` implementation that manages the WebSocket connection, audio buffering with ActivityStart/ActivityEnd framing, message mapping, image sending via `clientContent`, and function call orchestration.
- **`InternalLiveTransport.cs`** — Internal WebSocket transport (`Client`, `Live`, `AsyncSession`) handling connection lifecycle, credential headers, binary frame send/receive, and graceful disposal with close timeouts.
- **`InternalLiveTypes.cs`** — Internal JSON-serializable types for the Vertex AI Live API protocol (client messages, server messages, blobs, function calls, schemas, etc.).
- **`InternalLiveJsonContext.cs`** — Source-generated `JsonSerializerContext` for AOT-safe serialization of all Live API types.
- **`BuildIRealtimeClientTest.cs`** — 42 unit tests covering client construction, session config mapping, audio commit flow, message mapping, function call handling, disposal, and edge cases.
- **`AotJsonContextTest.cs`** — 4 unit tests verifying AOT source-gen coverage, nested type discovery, and serialization round-trips.

### Modified Files
- **`VertexAIExtensions.cs`** — Added `BuildIRealtimeClient()` / `BuildIRealtimeClientAsync()` extension methods on `PredictionServiceClientBuilder`.
- **`PredictionServiceChatClient.cs`** — Eliminated JSON round-tripping for tool arguments/results to improve AOT compatibility (direct `Struct` ↔ dictionary conversion).
- **`BuildIChatClientTest.cs`** — Updated tests reflecting the chat client tool-handling changes.
- **`Google.Cloud.VertexAI.Extensions.csproj`** — Version bumped to `1.0.0-beta08`; added `Microsoft.Bcl.AsyncInterfaces` dependency for `netstandard2.0`/`net462`.
- **`docs/history.md`** — Release notes for `1.0.0-beta08`.

## Features

- ✅ **Audio streaming** — Append/commit pattern with automatic frame splitting (32KB max), explicit ActivityStart/ActivityEnd framing
- ✅ **Manual activity detection** — Vertex AI does not support `audioStreamEnd`; automatic activity detection is always disabled in favor of explicit ActivityStart/ActivityEnd framing that reliably triggers model responses
- ✅ **Text conversations** — Send text messages and receive text/audio responses
- ✅ **Image understanding** — Images sent via `clientContent` with `inlineData` for proper multimodal conversation context
- ✅ **Function calling** — Full tool invocation support with the `FunctionInvokingRealtimeSession` middleware; tool responses batched into a single `SendToolResponseAsync` call
- ✅ **Transcription** — Input and output audio transcription support
- ✅ **Thread-safe sends** — `SemaphoreSlim` serializes all WebSocket sends, safe for concurrent middleware + caller usage
- ✅ **Graceful disposal** — Race-safe dispose with proper exception handling for in-flight operations; WebSocket close with 5-second timeout
- ✅ **SetupComplete handshake** — `CreateSessionAsync` waits for the server's `SetupComplete` before returning
- ✅ **Credential resolution** — Supports ADC (Application Default Credentials), service account JSON (`JsonCredentials`), and explicit `GoogleCredential` — all automatically scoped with `cloud-platform` OAuth scope
- ✅ **AOT compatible** — Source-generated JSON serialization for all Live API types; chat client tool path also improved for AOT
- ✅ **Schema depth guard** — `ConvertJsonSchemaToGoogleSchema` enforces MaxDepth=32 to prevent stack overflow from deeply nested schemas
- ✅ **Usage metadata** — Token counts surfaced in `ResponseDone` messages, correctly handled even when `TurnComplete` and `UsageMetadata` arrive in the same server message

## Key Design Decisions

1. **Always manual activity detection** — Vertex AI does not support `audioStreamEnd` (confirmed by `dotnet-genai`'s `LiveConverters.cs`). When automatic activity detection is enabled server-side, the server ignores manual `ActivityStart`/`ActivityEnd` signals, leaving no way to trigger a response. The provider always forces `AutomaticActivityDetection.Disabled = true` and uses explicit `ActivityStart` → audio → `ActivityEnd` framing regardless of the user's `VoiceActivityDetection.Enabled` setting. The `AllowInterruption` option is still respected via `activityHandling`.

2. **Images via `clientContent`** — Static images are sent via `clientContent` with `inlineData` parts (proper conversation content) rather than `realtimeInput.video` (designed for streaming video frames). This ensures the model properly processes images as part of the conversation context.

3. **Tool response batching** — The MEAI `FunctionInvokingRealtimeSession` middleware sends separate `CreateConversationItem` per function result. Gemini expects all results in one `SendToolResponseAsync` call. The provider buffers results and flushes them as a single batch when `CreateResponse` arrives.

4. **TurnComplete suppression after tool responses** — After `SendToolResponseAsync`, the Gemini model automatically continues generating. The provider tracks this via `_lastSendWasToolResponse` and skips redundant triggers.

5. **SetupComplete handshake** — The `CreateSessionAsync` method drains the server's `SetupComplete` acknowledgment before returning, ensuring the session is fully ready before the caller sends audio or text.

6. **Audio buffer cap** — Audio appends are capped at 10 MB to prevent unbounded memory growth. Frames exceeding 32 KB are automatically split.

7. **Consistent with IChatClient** — The realtime provider follows the same patterns as `PredictionServiceChatClient`: same namespace, same builder extension methods (`BuildIRealtimeClient`/`BuildIRealtimeClientAsync`), same credential resolution, same `GetService` pattern exposing the underlying `Client` via `IServiceProvider`.

## Test Coverage

151 unit tests (42 new + 109 existing), covering:
- Client and session lifecycle (construction, disposal, idempotent dispose)
- All `BuildLiveConnectConfig` option combinations (modalities, voice, tools, transcription, VAD, max tokens)
- Audio commit flow (ActivityStart/ActivityEnd framing, chunking, empty buffer handling)
- All server message mapping (audio, text, transcription, function calls, usage, GoAway, turn complete)
- Function call handling (single/multiple results, batching, null args, ID generation)
- Edge cases (null inputs, empty buffers, concurrent dispose, exception swallowing)
- Media trigger flow (image → CreateResponse → clientContent with turnComplete)
- AOT source-gen context coverage, nested type discovery, and round-trip correctness
- Chat client tool argument/result direct conversion (replaces JSON round-trip tests)

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
I'll get to this sometime Tuesday/Wednesday. Thanks!

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeSession.cs [hunk 1]
```diff
@@ -0,0 +1,944 @@
...
+            case IEnumerable enumerable when value is not string:
+                var list = new List<object?>();
+                foreach (object? item in enumerable)
+                {
+                    list.Add(NormalizeToolPayload(item, depth + 1));
+                }
+                return list;
+
+            default:
+                return value.ToString();
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeSession.cs]: Fixed.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+            {
+                case WebSocketState.Connecting:
+                    throw new InvalidOperationException("Cannot receive data while the WebSocket is still connecting.");
+                case WebSocketState.Open:
+                    break;
+                default:
+                    return null;
+            }
+
+            byte[] buffer = new byte[4096];
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs]: Addressed.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+                }
+            } while (!result.EndOfMessage);
+
+            if (messageStream.Length == 0)
+            {
+                return null;
+            }
+
+            string json = Encoding.UTF8.GetString(messageStream.ToArray());
+            return JsonSerializer.Deserialize(json, LiveJsonContext.Default.LiveServerMessage);
```
[the remark this answers is not in the document]
Comment (tarekgh) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs]: Addressed.

**Comment by tarekgh** [comment]
Thanks so much for your help on this! As noted in the description, there are a few additional changes in the SDK itself to support AOT compatibility.

I’d also really appreciate any help in getting visibility or support for this PR as well: https://github.com/googleapis/dotnet-genai/pull/256.

CC @jeffhandley @stephentoub

File: apis/Google.Cloud.VertexAI.Extensions/docs/history.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
 # Version history
 
+## Version 1.0.0-beta08, unreleased
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/docs/history.md]: This will be generated on release based on the commit name and messages. Use https://www.conventionalcommits.org/en/v1.0.0/, e.g. `feat: `.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Version>1.0.0-beta07</Version>
+    <Version>1.0.0-beta08</Version>
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj]: Also, this will be automatically changed by the release process.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveJsonContext.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.GenAI.Types;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Google.GenAI;
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveJsonContext.cs]: Use the right namespace.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+using System.IO;
+using System.Net.WebSockets;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.GenAI;
+
+internal sealed class Client : IAsyncDisposable
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/InternalLiveTransport.cs:27]: Can't we use the generated client here? Se the other extensions.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
I've done a quick review and left a few comments.

For now, we won't be supporting a handwritten client, even if that's required for AoT. We want to keep this library lightweight and as a simple wrapper of the generated Google.Cloud.AiPlatform.V1.

We have on this year's roadmap looking into making the generated code AoT compatible and at that point, Google.Cloud.VertexAI.Extensions will benefit from it.

**Comment by tarekgh** [comment]
Thanks for the review, @amanda-tarafa! I'll address the inline comments (history.md, version bump, namespace); those are straightforward.

Regarding the architectural concern about the handwritten client, I want to explain why the Realtime provider can't follow the same pattern as the other extensions. Please correct me if I got anything wrong here. I may be missing context about the generated SDK's roadmap or capabilities.

How the other extensions work: `ChatClient`, `EmbeddingGenerator`, and `ImageGenerator` all call `builder.Build()` to get a generated `PredictionServiceClient`, then delegate to its gRPC methods (`GenerateContentAsync`, `StreamGenerateContent`, `PredictAsync`). I understand the expectation is for all extensions to follow this pattern.

Why the Realtime provider can't do this: The Vertex AI Live API uses a completely different transport and service. It's a WebSocket-based bidirectional streaming endpoint on a service called LlmBidiService:

```
 wss://aiplatform.googleapis.com/ws/google.cloud.aiplatform.v1beta1.LlmBidiService/BidiGenerateContent
```

This service has no generated client; there is no `LlmBidiServiceClient` in `Google.Cloud.AIPlatform.V1` or `V1Beta1`. The proto definitions for `LlmBidiService` are not published in the protobuf sources that the SDK is generated from, so there is nothing to wrap.

Could the generated bidi-streaming gRPC methods work instead? The generated `PredictionServiceClient` does have bidi-streaming RPCs (`StreamDirectPredict`, `StreamDirectRawPredict`, `StreamingPredict`, `StreamingRawPredict`). I looked into whether any of these could serve as the transport for the Live API. They can't; these are methods on `PredictionService`, not `LlmBidiService`. They route to different server-side implementations. For example, `StreamDirectRawPredict` carries arbitrary `ByteString` payloads, but sending Live API protocol messages through it wouldn't reach the Live API backend. These RPCs are designed for raw tensor/model inference, not the Live API's structured protocol (setup handshake, activity detection, audio framing, function calling, etc.). Google's own documentation confirms WebSocket is the only supported transport for the Live API.

This is the same approach used in the [dotnet-genai sister PR](https://github.com/googleapis/dotnet-genai/pull/256), which also implements a WebSocket-based transport for the same endpoint.

In summary: the handwritten WebSocket transport isn't a choice to bypass the generated client; it's a necessity because no generated client exists for this API surface, and the server only accepts WebSocket connections for the Live API. Happy to discuss the best path forward.

**Comment by amanda-tarafa** [comment]
@tarekgh I ran the checks and they are failing because of some dependencies we had to update. You'll need to rebase on main.

**Comment by tarekgh** [comment]
@amanda-tarafa I have rebased and should be ready now. Thanks!

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj [hunk 8]
```diff
@@ -10,6 +10,7 @@
   <ItemGroup>
     <PackageReference Include="ConfigureAwaitChecker.Analyzer" PrivateAssets="All" />
     <PackageReference Include="Google.Cloud.AIPlatform.V1" VersionOverride="[3.63.0, 4.0.0)" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" VersionOverride="10.0.4" Condition="'$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'net462'" />
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj]: Remove the conditions as they are redundant.

Comment (amanda-tarafa) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj]: Also, you should get these dependencies transitively, any reason why you added this one explicitly here?

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+        {
+            gc = gc.CreateScoped(_builder.Scopes ?? PredictionServiceClient.DefaultScopes);
+            if (!string.IsNullOrWhiteSpace(_builder.QuotaProject))
+            {
+                gc = gc.CreateWithQuotaProject(_builder.QuotaProject);
+            }
+            credential = gc;
+        }
+
+        GenAIHttpOptions httpOptions = new()
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs]: Remove all the GenAI references.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/VertexAIExtensions.cs [hunk 10]
```diff
@@ -74,6 +74,58 @@ public static async Task<IChatClient> BuildIChatClientAsync(
...
+    public static Task<IRealtimeClient> BuildIRealtimeClientAsync(
+        this PredictionServiceClientBuilder builder,
+        string? defaultModelId = null, IServiceProvider? provider = null,
+        CancellationToken cancellationToken = default)
+    {
+        if (cancellationToken.IsCancellationRequested)
+        {
+            return Task.FromCanceled<IRealtimeClient>(cancellationToken);
+        }
+        return Task.FromResult(builder.BuildIRealtimeClient(defaultModelId, provider));
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/VertexAIExtensions.cs]: Let's create a client builder type specifically for the live client, and encapsulate all the creation logic there, you can extend GAX's ClientBuilderBase. You are really not wrapping a prediction service client here, so using the builder is misleading.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+        if (credential is null && !string.IsNullOrWhiteSpace(_builder.JsonCredentials))
+        {
+            credential = GoogleCredential.FromJson(_builder.JsonCredentials);
+        }
+#pragma warning restore CS0618
+
+        // Ensure the credential has the cloud-platform scope required by Vertex AI.
+        // The gRPC channel builder handles this automatically, but the WebSocket live
+        // path needs it applied explicitly.
+        if (credential is GoogleCredential gc)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs]: When you implement a client builder for the live client, a lot of this will be encapsulated there, and done by the builder itself.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
I haven't done an in-depth review but I do have some fundamental change requests regarding the client builder.

Also, I prefer to simplify this PR by removing all the AoT related changes. Some of our dependencies are not AoT so this package won't really be AoT until we've done some major, upstream changes on our side.

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj (hunk 8 above)
Comment (tarekgh) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions.csproj]: Done

File: apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs (hunk 9 above)
Comment (tarekgh) [reply apis/Google.Cloud.VertexAI.Extensions/Google.Cloud.VertexAI.Extensions/PredictionServiceRealtimeClient.cs]: Done

**Comment by tarekgh** [comment]
@amanda-tarafa I have addressed all your comments.

**Comment by amanda-tarafa** [comment]
@tarekgh I've sent an internal email with more details.

I'll be closing this PR now as this is not a wrapper on a generated service, and requires heavy handwritten involvement including for future fields, types etc. This is a model we are not supporting on this repo at the moment.

Note that the PR itself is fine, it's the nature of the service and the fact that it's not generated that prevents us from moving forward.
